src/simulator/RangeSensor.tsx

```diff
@@ -120,8 +125,15 @@ const RangeSensor = ({
               value={value}
               textAlign="center"
               mt="-8"
-              ml={-valueText.length / 2 + "ch"}
+              ml={
+                valuePercent < 25
+                  ? 0
+                  : valuePercent > 75
+                  ? -valueText.length + "ch"
+                  : -valueText.length / 2 + "ch"
+              }
```

Comment:
Open to alternative suggestions to keep the labels from disappearing outside the container.

Reply:
This feels much better [`8c06fd5`](https://github.com/microbit-foundation/python-editor-next/pull/923/commits/8c06fd57656f288b999ab5843ebfb7149eef45d3).